### PR TITLE
feat: Home page with categories listed

### PR DESCRIPTION
**Pull request by partha8**
## Files to check
- [src/components/Navbar/Navbar.tsx](https://github.com/partha8/algern-quiz/compare/home?expand=1#diff-3fb9bb17462f833c508f7e5c5d2d9273592b779c587257b5ea04cb1577d32091)
- [src/pages/Home/Home.tsx](https://github.com/partha8/algern-quiz/compare/home?expand=1#diff-432cf2dcda728ba0eb01dcfad6212ae68c0dd153482d759eb5571b8b71ac0e60)
- [src/redux/reducers/quizSlice.ts](https://github.com/partha8/algern-quiz/compare/home?expand=1#diff-92227e64a07e2d71ae5bec111040acba8ca421481041a2eef2e0f5f35f865f16)
- [src/redux/store.tsx](https://github.com/partha8/algern-quiz/compare/home?expand=1#diff-e563919846cf6846244c88c97aaa3d4a191afd7706d8c9108e7300ec23ef7758)
- [src/types/quiz.types.ts](https://github.com/partha8/algern-quiz/compare/home?expand=1#diff-b4c4a87e11dbc2c982df41042e032d064446449ed313929e6bf66d745ebb1e85)

## Preview
image

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 const root = ReactDOM.createRoot(
-  document.getElementById('root') as HTMLElement
+  document.getElementById("root") as HTMLElement
 );
 root.render(
   <React.StrictMode>
-    <App />
+    <Provider store={store}>
+      <App />
+    </Provider>
```
Comment (Dev-AkshatGupta) [inline src/index.tsx:16]: Suggestion : Take care of the indentation

File: src/redux/hook.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
+import type { RootState, AppDispatch } from "./store";
+
+// Use throughout your app instead of plain `useDispatch` and `useSelector`
+export const useAppDispatch: () => AppDispatch = useDispatch;
+export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
```
Comment (Dev-AkshatGupta) [inline src/redux/hook.ts:6]: Great that you made custom hooks

File: src/redux/reducers/quizSlice.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+);
+
+const quizSlice = createSlice({
+  name: "quiz",
+  initialState,
+  reducers: {},
+  extraReducers(builder) {
+    builder
+      .addCase(getCategories.pending, (state, action) => {
+        console.log("pending");
```
Comment (Dev-AkshatGupta) [inline src/redux/reducers/quizSlice.ts:48]: Remove consoles once you are done with them

File: src/redux/reducers/quizSlice.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      let categories = [];
+      categories = snapShot.docs.map((doc) => {
+        const data = doc.data();
+        data["id"] = doc.id;
+        return data as Categories;
+      });
+      return categories;
+    } catch (error) {
+      console.error(error);
+      return thunkAPI.rejectWithValue(error);
```
Comment (Dev-AkshatGupta) [inline src/redux/reducers/quizSlice.ts:36]: Suggestion : You can destructure the reject with value

**Review by Dev-AkshatGupta — APPROVED** [review]
Provided some suggestions everything else was great.
Things I liked are:

1. You are using typescript
2. you are using rtk for state's

File: src/redux/reducers/quizSlice.ts (hunk 3 above)
Comment (partha8) [reply src/redux/reducers/quizSlice.ts:48]: In the other branches, i have removed them.

**Pull request merged**